server/websocket/src/connection.rs

```diff
@@ -325,10 +325,18 @@ where
                             // Send errors back to the client as GQL_DATA
                             match e {
                                 SubscriptionError::GraphQLError(e) => {
-                                    let result = Arc::new(QueryResult::from(e));
-                                    let msg =
-                                        OutgoingMessage::from_query_result(err_id.clone(), result);
-                                    error_sink.unbounded_send(msg.into()).unwrap();
+                                    // Don't bug clients with transient `TooExpensive` errors,
+                                    // simply skip updating them
+                                    if e.iter().all(|err| {
+                                        !matches!(err, QueryExecutionError::TooExpensive)
```

Comment:
This is better written as `if !e.iter.any(|err| { matches!(...`.

Reply:
Yep; I though `!matches!` looked funny, but didn't think of writing it that way. Fixed that and rebased to latest master